homeassistant/components/plant/__init__.py

```diff
@@ -123,6 +132,20 @@ async def async_setup(hass, config):
         entities.append(entity)
 
     await component.async_add_entities(entities)
+
+    @callback
+    def websocket_handle_get_config(hass, connection, msg):
+        """Handle getting a plant config."""
+
+        entity_id = msg["entity_id"].split(".")[1]
+        plant_config = config[DOMAIN][entity_id]
```

Comment:
`config[DOMAIN][entity_id]` does not work for everyone as the key in the configuration does not need to be the entity_id.

Example Configuration:
```
plant:
  Callisia Repens:
    sensors:
      moisture: sensor.callisia_repens_moistu
      conductivity: sensor.callisia_repens_co
      brightness: sensor.callisia_repens_ligh
      temperature: sensor.callisia_repens_tem
      battery: sensor.callisia_repens_battery
```

For this the resulting entity_id would be `callisia_repens`.

Reply:
Hum.  So the issue is to go from an entity_id back to the key in the config?
One _could_ potentially loop through the plants and do a `generate_entity_id` on all of them, and match on that, but it does not sound like a good solution...

Reply:
exactly; I was looking for a way to get the plant component class from hass by it's entityId but did not find anything helpful. In generell having a way to get the information from the plant itself and not from the initial configuration sounds better.

Reply:
But the question is - what will you use to query the websocket api?
entity_id or the name from the config?
If we want to query by entity_id we can simply `hass.states.get(entity_id)`, and find the relevant max/mins etc. from that instead.
It will be a few more lines of code, but nothing crazy, I think.

Reply:
And... or... maybe we should add a service in addition to the websocket?
```
service: plant.get_limits
  entity_id: plant.callisia_repens
```
Would that make sense?
(It would be in another PR, but basically the same code)

Reply:
One option would be to establish the mapping in `plant.async_added_to_hass()` with `self.hass.data[DOMAIN]["configs"][self.entity_id] = self._config`.

Reply:
Good idea

Reply:
Wouldn't `hass.states.get(entity_id)` give you the state? The limits are not part of the state (Adding limits there was the way i tried in my pr 😄).

Reply:
Bud do we have any idea how to map? As entities could be renamed inside hass i don't see any way that did not query the mapping from hass.

Reply:
I believe entity_ids for plant component are static at present. That will change with config entries at a later point but should hold until then.

So, with the above snippet in `async_added_to_hass()`, you can access the plant config in the websocket callback with` plant_config = hass.data[DOMAIN]["configs"][entity_id]` and return that to the front-end. You will need to initialise `hass.data[DOMAIN]` at the beginning of `async_setup()`.

Sorry if I have misunderstood your question.

Reply:
Pushed an update now.